quarkus-solace-client/deployment/src/main/java/com/solace/quarkus/deployment/SolaceProcessor.java

```diff
@@ -59,21 +59,24 @@ ExtensionSslNativeSupportBuildItem ssl() {
     @Record(ExecutionTime.RUNTIME_INIT)
     ServiceStartBuildItem init(
             SolaceConfig config, SolaceRecorder recorder,
-            ShutdownContextBuildItem shutdown, BuildProducer<SyntheticBeanBuildItem> syntheticBeans) {
+            ShutdownContextBuildItem shutdown, BuildProducer<SyntheticBeanBuildItem> syntheticBeans,
+            BuildProducer<AdditionalBeanBuildItem> additionalBeanBuildItemBuildProducer) {
 
         Function<SyntheticCreationalContext<MessagingService>, MessagingService> function = recorder.init(config, shutdown);
 
+        additionalBeanBuildItemBuildProducer.produce(AdditionalBeanBuildItem.unremovableOf(OidcProvider.class));
```

Comment:
@ozangunalp is there a way we can produce this bean only if oidc properties are configured by user. I see some warnings/errors thrown by oidc-client library when no configuration is available(in case of basic authentication or any other type of authentication). This is not blocking message flow.

Reply:
I think the easier option is to make it do nothing when oauth authentication is not configured.